figures/S5.supplemental_figure5/scripts/S5.r

```diff
@@ -0,0 +1,1448 @@
+
+
+## function to process the data for visualization
+process_subset_data <- function(df){
+    # read in the data
+    # data <- read.csv(data_path, header = TRUE, sep = ",", stringsAsFactors = FALSE)
+    # get the basename of the files
+
+    data <- df %>%
+        dplyr::arrange(desc(abs(coefficients))) %>%
+        tidyr::separate(
+            feature_names,
+            into = c(
+                "compartment",
+                "feature_group",
+                "measurement",
+                "channel",
+                "parameter1",
+                "parameter2"
+            ),
+            sep = "_",
+            remove = FALSE
+        ) %>%
+        dplyr::mutate(channel_cleaned = channel) %>%
+        dplyr::arrange(desc(abs(coefficients)))
+
+    # Clean channel for visualization
+    data$channel_learned <- dplyr::recode(data$channel,
+            "CorrDNA" = "nuclei",
+            "CorrMito" = "Mito",
+            "CorrER" = "ER",
+            "CorrGasdermin" = "gasdermin",
+            "CorrPM" = "PM",
+            .default = "other",
+            .missing="other"
+    )
+    data <- data %>%
+        dplyr::group_by(feature_group, channel_learned, compartment) %>%
+        dplyr::slice_max(order_by = coefficients, n = 1)
+    return(data)
+}
+
+
+plot_coeffs <- function(df, cytokine, shuffle){
+
+# plot the data
+coef_gg <- (
+    ggplot(df, aes(x = channel_learned, y = feature_group))
+    + geom_point(aes(fill = abs(coefficients)), pch = 22, size = 5.75)
+    + facet_wrap("~compartment", ncol = 3)
+    + theme_bw()
+    + scale_fill_continuous(
+        name="Top Abs. val\ntreatment\nlinear model\ncoefficient",
+        low = "darkblue",
+        high = "yellow",
+    )
+    + xlab("Channel")
+    + ylab("Feature")
+
+    + figure_theme
+    + theme(
+        axis.text.x = element_text(angle = 45, hjust = 1, size = 14),
+    )
+    # rotate x axis labels
+    + theme(axis.text.x = element_text(angle = 45, hjust = 1))
+    + ggtitle(paste0("Top Abs. val treatment ElasticNet coefficients for \n",cytokine,shuffle," model"))
+    + theme(plot.title = element_text(hjust = 0.5))
+    )
+    return(coef_gg)
+}
+
+
+# get all files in a directory
+files <- list.files(path = input_file_path, pattern = "*.csv", full.names = TRUE)
+
+# create empty list (mimics a dictionary )
+nested_list <- list(
+        filename = list(),
+        cytokine = list(),
+        shuffle = list()
+    )
+
+for (i in files){
+    filename <- basename(i)
+    # split the string at the first _
+    filename <- strsplit(filename, "_", fixed = TRUE)[[1]]
+    cytokine <- filename[1]
+    shuffle <- filename[2]
+    nested_list$filename <- c(nested_list$filename, i)
+    nested_list$cytokine <- c(nested_list$cytokine, cytokine)
+    nested_list$shuffle <- c(nested_list$shuffle, shuffle)
+}
+
+
+cytokine_list <- c(
+    "IL-1beta",
+    "TFalpha",
+    "CCL24",
+    "IL-18",
+    "Osteopontin(OP)",
+    "CCL13",
+    "IL-2",
+    "IL-6",
+    "CCL4")
```

Comment:
I _think_ if you define this earlier you can use it in your code to define `agg_df` (line 198).

Reply:
removed